test/variational/test_exact_state.py

```diff
@@ -122,13 +130,13 @@ def test_qutip_conversion(vstate):
     np.testing.assert_allclose(q_obj.data.todense(), ket.reshape(q_obj.shape))
 
 
-def test_derivatives_agree():
+@pytest.mark.parametrize("machine", machines.values())
+def test_derivatives_agree(machine):
     err = 1e-3
     g = nk.graph.Hypercube(length=10, n_dim=1, pbc=True)
```

Comment:
I know it wasn't you, but let's reduce slightly the size so the tests run faster, and use Chain which is equivalent but more readable
```suggestion
    g = nk.graph.Chain(length=8, pbc=True)
```